files/aws/usr/bin/update_netplan_to_mac_address

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 by Delphix. All rights reserved.
+#
+
+# 1. This script is applicable only for Upgrade scenarios in AWS deployments
+# 2. The fix for DLPX-75209 shifts to use mac addresses instead of canonical network
+# interface names. To handle this change for upgrades, we save a map from interface
+# names to their mac addresses. Post reboot, this script will be called as part of
+# the PreExecStart directive in systemd-networkd service.
+# 3. The script takes the map and matches the keys with the existing interface names
+# in the netplan file, and for each such match, replaces the interface name with the
+# corresponding mac address stored in the map file
+
+INTERFACE_TO_MACADDRESS_MAP_FILE="/etc/interface_to_macaddress_map.out"
+DLPX_NETPLAN_DIR="/etc/netplan/"
+DLPX_NETPLAN_NAME="10-delphix.yaml"
+SYSFS_NET_DIRECTORY="/sys/class/net"
+
+function die() {
+	echo "$(basename "$0"): $*" >&2
+	exit 1
+}
+
+# We want to process the map file only when we boot into the new version post upgrade, and not during
+# any other scenario where the systemd-networkd service is restarted. This necessitates check for the
+# following conditions which are valid only we boot into the new version post upgrade.
+# 1. Existence of the map file, which is generated during the upgrade processing.
+# 2. Existence of mac address based network interface filenames under the /sys/class/net heirarchy.
+# 3. The netplan file /etc/netplan/10-delphix.yaml should not contain mac address based interface names.
+if [[ -e "$INTERFACE_TO_MACADDRESS_MAP_FILE" ]] &&
+	find "$SYSFS_NET_DIRECTORY" -regextype egrep -regex '.*enx[a-z0-9]{12}$' | read -r &&
+	! grep -E 'enx[a-z0-9]{12}' "$DLPX_NETPLAN_DIR""$DLPX_NETPLAN_NAME"; then
+	# save the existing netplan file before updating it, could be handy for debugging
+	cur_timestamp=$(date +%y%m%d_%h%m%s)
+	cp -p "$DLPX_NETPLAN_DIR""$DLPX_NETPLAN_NAME" /var/tmp/"$DLPX_NETPLAN_NAME"_"$cur_timestamp"
+
+	# Update the netplan file with MAC based interface names
+	while IFS= read -r line; do
+		ifname=$(echo "$line" | awk '{print $1}')
+		macname=$(echo "$line" | awk '{print $2}' | tr -d ":")
+		macname="enx""$macname"
+		if grep -q "$ifname:" "$DLPX_NETPLAN_DIR""$DLPX_NETPLAN_NAME"; then
+			echo "Modifying $ifname to $macname in the netplan file" >>/var/log/syslog 2>&1
```

Comment:
I think `logger` is preferred to writing to the system logic.. can we use that instead?

Reply:
Sure. I will study the usage of logger command and update.

Reply:
As suggested in another comment, I have updated all the echo statements to write to STDOUT to enable all the output to be visible in one place, i.e. output of systemd journal file.